t/backup_expiration.t

```diff
@@ -119,38 +119,49 @@ describe "HTFeed::BackupExpiration" => sub {
-      ok(!zip_deleted($storage), "new ($storage->{timestamp}) zip left intact");
+      ok(!mets_deleted($storage), "single ($storage->{timestamp}) mets left intact");
+      ok(!zip_deleted($storage), "single ($storage->{timestamp}) zip left intact");
     };
 
-    it "deletes old versions when there is a new one" => sub {
+    it "does not do anything with a single old version and single new one" => sub {
+      my $old_storage = prepare_storage($vars{storage_name}, old_random_timestamp());
+      my $new_storage = prepare_storage($vars{storage_name}, new_random_timestamp());
+      my $exp = HTFeed::BackupExpiration->new(storage_name => $vars{storage_name});
+      $exp->run();
+      my $deleted = count_deleted_objects('test', 'test', $old_storage->{timestamp});
+      is($deleted, 0, 'old object is not deleted');
+      ok(!mets_deleted($old_storage), "old ($old_storage->{timestamp}) mets left intact");
+      ok(!zip_deleted($old_storage), "old ($old_storage->{timestamp}) zip left intact");
+
+      $deleted = count_deleted_objects('test', 'test', $new_storage->{timestamp});
+      is($deleted, 0, 'new object is not deleted');
+      ok(!mets_deleted($new_storage), "new ($new_storage->{timestamp}) mets left intact");
+      ok(!zip_deleted($new_storage), "new ($new_storage->{timestamp}) zip left intact");
+    };
+
+    it "does not delete old versions when there is nothing new" => sub {
       my @old_versions;
       foreach my $n (1 .. 2) {
         my $storage = prepare_storage($vars{storage_name}, old_random_timestamp());
         push @old_versions, $storage;
       }
-      my $storage = prepare_storage($vars{storage_name}, new_random_timestamp());
       my $exp = HTFeed::BackupExpiration->new(storage_name => $vars{storage_name});
       $exp->run();
-      my $deleted = count_deleted_objects('test', 'test');
-      is($deleted, 2, 'two old objects marked with feed_backups.deleted=1');
-      $deleted = count_deleted_objects('test', 'test', $storage->{timestamp});
-      is($deleted, 0, 'newest object still feed_backups.deleted=0');
       foreach my $old_storage (@old_versions) {
-        ok(mets_deleted($old_storage), "old ($old_storage->{timestamp}) mets deleted");
-        ok(zip_deleted($old_storage), "old ($old_storage->{timestamp}) zip deleted");
+        my $deleted = count_deleted_objects('test', 'test', $old_storage->{timestamp});
+        is($deleted, 0, 'old object is not marked feed_backups.deleted');
+        ok(!mets_deleted($old_storage), "old ($old_storage->{timestamp}) mets left intact");
+        ok(!zip_deleted($old_storage), "old ($old_storage->{timestamp}) zip left intact");
       }
-      ok(!mets_deleted($storage), "new ($storage->{timestamp}) mets left intact");
-      ok(!zip_deleted($storage), "new ($storage->{timestamp}) zip left intact");
     };
 
-    it "keeps the newest old version" => sub {
+    it "deletes the oldest old version when there is a new one" => sub {
```

Comment:
consider as a description "deletes old versions after 6 months after a newer version has been backed up"

Reply:
I'm keeping separate tests with and without "new" versions since that's been the crux of this latest round of confusion. Using the descriptions "deletes the oldest old versions when there are no new ones" and "deletes the oldest old versions when there are new ones". Since I seem to have standardized on "old" to mean > 6mo and "new" to mean <6mo... and where needed "newer" to substitute for "newest old".